### PR TITLE
Support jest 28

### LINKED ISSUE #467: Support jest 28
[Jest 28](https://github.com/facebook/jest/blob/HEAD/CHANGELOG.md#2800) has been released two days ago.  We upgraded our project following the official [migration guide](https://jestjs.io/docs/upgrading-to-jest28).

Currently, we are using the following versions:
```js
  "devDependencies": {
    "@vue/vue3-jest": "27.0.0",
    "babel-jest": "27.5.1",
    "jest": "27.5.1",
  },
  "dependencies": {
    "@jest/globals": "27.5.1",
    "vue": "3.2.33",
  }
```

After upgrading to v28, many of our tests are failing with the following error message:
```
Test suite failed to run

    TypeError: babelJest.getCacheKey is not a function

      at Object.getCacheKey (node_modules/@vue/vue3-jest/lib/index.js:13:19)
      at ScriptTransformer._getCacheKey (node_modules/@jest/transform/build/ScriptTransformer.js:278:41)
      at ScriptTransformer._getFileCachePath (node_modules/@jest/transform/build/ScriptTransformer.js:349:27)
      at ScriptTransformer.transformSource (node_modules/@jest/transform/build/ScriptTransformer.js:590:32)
      at ScriptTransformer._transformAndBuildScript (node_modules/@jest/transform/build/ScriptTransformer.js:760:40)
      at ScriptTransformer.transform (node_modules/@jest/transform/build/ScriptTransformer.js:817:19)
```

I have not found any hints in this repo when vue-jest will support jest v28. Are there any plans or a roadmap?

### PR DESCRIPTION
**Pull request by valentinpalkovic**
fixes #467

[BREAKING CHANGE]: 
- Drop support for Jest 27
- Drop support for Node 10 (Drop Node 10 support as Jest 28 dropped it as well)
- Drop support for Typescript < 4.3 (Jest 28 requires Typescript >= 4.3)

### CODE REVIEW

**Comment by lmiller1990** [comment]
Nice job, I'll take a look at this soon.

File: README.md [hunk 1]
```diff
@@ -10,8 +10,8 @@ Since we need to support a variety of Vue and Jest versions, vue-jest doesn't fo
 | ----------- | ----------------- | ---------------- |
 | Vue 2       | Jest 26 and below | `vue-jest@4`     |
 | Vue 3       | Jest 26 and below | `vue-jest@5`     |
-| Vue 2       | Jest 27           | `@vue/vue2-jest` |
-| Vue 3       | Jest 27           | `@vue/vue3-jest` |
+| Vue 2       | Jest 28           | `@vue/vue2-jest` |
+| Vue 3       | Jest 28           | `@vue/vue3-jest` |
```
Comment (nogic1008) [inline README.md:14]:
```suggestion
| Vue 2       | Jest 26 and below | `vue-jest@4`        |
| Vue 3       | Jest 26 and below | `vue-jest@5`        |
| Vue 2       | Jest 27 and above | `@vue/vue2-jest@xx` |
| Vue 3       | Jest 27 and above | `@vue/vue3-jest@xx` |

**xx**: Major version of Jest
```

**Review by nogic1008 — CHANGES_REQUESTED** [review]
Code fixes is LGTM, but docs need to update for clarify versioning.

**Comment by lmiller1990** [comment]
Once the docs is updated we can merge and release the Jest 28 support. 🙏

**Comment by lmiller1990** [comment]
Nice job @valentinpalkovic. Please give me a day or two to get this merged and released, since we need to also create the legacy 27.x branch before merging, and a few other things. I'm glad supporting Jest 28 was quite straight forward.

**Review by nogic1008 — APPROVED** [review]

**Comment by ScottAwesome** [comment]
@lmiller1990 any chance of landing this in the next few days?

**Comment by lmiller1990** [comment]
I will probably be busy until the weekend, so it should be done by the start of next week.

**Comment by lmiller1990** [comment]
I will do this right now.

**Review by lmiller1990 — APPROVED** [review]

**Comment by lmiller1990** [comment]
It's out, please give it a try: https://github.com/vuejs/vue-jest/releases/tag/v28.0.0

**Pull request merged**